File: RB101/ruby-small-problems/easy6/2.rb
VOWELS = ["a", "e", "i", "o", "u"]
def remove_vowels(arr)
  arr.map do |string|
    string.chars.reject { |char| VOWELS.include?(char.downcase) }.join
  end
end

# p remove_vowels(%w(green YELLOW black white))

=begin
P - take an array and return it with all vowel characters removed.
E -
remove_vowels(%w(green YELLOW black white)) == %w(grn YLLW blck wht)
case insensitive
input: array
output: array
D - array
A -
initialize constant variable of vowels
iterate over array
if vowels includes current character,
  next
else
  return current character
C -
=end


# test cases
p remove_vowels(%w(abcdefghijklmnopqrstuvwxyz)) == %w(bcdfghjklmnpqrstvwxyz)
p remove_vowels(%w(green YELLOW black white)) == %w(grn YLLW blck wht)
p remove_vowels(%w(ABC AEIOU XYZ)) == ['BC', '', 'XYZ']

File: JS210/js-exercises/string-text-processing/2.js
function removeVowels(arr) {
  const VOWELS = ["a", "e", "i", "o", "u"];
  let result = [];
  arr.forEach((element) => {
    VOWELS.forEach((vowel) => {
      if (element.includes(vowel)) {
        element = element.split(vowel).join("");
      } else if (element.includes(vowel.toUpperCase())) {
        element = element.split(vowel.toUpperCase()).join("");
      }
    });

    result.push(element);
  });

  console.log(result);
}

removeVowels(["abcdefghijklmnopqrstuvwxyz"]); // ["bcdfghjklmnpqrstvwxyz"]
removeVowels(["green", "YELLOW", "black", "white"]); // ["grn", "YLLW", "blck", "wht"]
removeVowels(["ABC", "AEIOU", "XYZ"]); // ["BC", "", "XYZ"]
